### DISCUSSION TITLE
AI Repainter Feature

### DISCUSSION THREAD

**Discussion by 3Diva** [upscayl/upscayl#623] [Q&A]
Is there a way to get Upscayl to behave a bit like the Krea image enhancer - in that it can add more detail to the image it upscales?

**Comment by NayamAmarshe** [comment]
Upscayl already does that. What is Krea doing differently?

**Reply by 3Diva** [reply]
It may create a tiny bit of detail but not much at all, it's barely noticeable if at all. Krea enhances the image and adds in more detail. I was just wondering if there is a way to get Upscayl to do that too - maybe a way to change the denoising strength. I'm using the local install GUI from https://www.upscayl.org/ and it doesn't seem to have much in the way of customization or settings adjustment. Was just hoping there was a way to get it to behave more like an image enhancer. If not that's fine, just seems like it doesn't hurt to ask.

**Comment by GRwald** [comment]
perhaps a latent upscale perhaps? or something like: https://github.com/Coyote-A/ultimate-upscale-for-automatic1111 .. but it seems to use the same teach, only in tiles... but because in some way using it in comfyui or automatic1111 it adds generation on top? I am experimentating with it to find something similar to both but localy with comfyui...  would be great though it`s a ginat leap foreward! 

variation: https://www.youtube.com/watch?v=iLODwfjd-Ys

**Reply by 3Diva** [reply]
@GRwald I haven't used ComfyuI much but will look into that. I'd love to see a local install image enhancer/upscaler - was hoping there was some "under the hood" tweaking that could be done to Upscayl to get it to behave a bit more like an enhancer while it upscales. It's a wonderful upscaler, if you're already 100% satisfied with the image and it's already highly detailed it's great and does a wonderful job, but with lower res images it doesn't help them as much as I'd like. Was hoping that there could be adjustments made to it to get it to work more like Krea or Magnific.

**Reply by GRwald** [reply]
I did manage to get a lot further and I did have great succes! (insert borad meme)

Ik gues the trick is the diffusion process. I started with and image that is smaller then 1500 pixels , actually needed to do a 50% scaledown of my art, and later on upscale it again with 50%, bet the results where staggering! much more detailed.

During the diffusion process I used a lora to add detail (it is in the comfyui workflow below), that was the most important. A low denoise of around 25. and an realEsrgan of 2x not the very best or brightest.. but it`s the 2x times that does the trick. with every 4x times the endresults were too grainy textured.. but now just enough... there is a controlnet tile (not yet available for SDXL) in order to get rid of most of the tiles.... so for my workflow I managed with SD 1.5, very nice!
Nice trick i got from a youtuber that it  calculated the desired end resolutions from the image you feed it. So just put in an image and it will sort itself out... and I made a switch for batch and single image :) 

Comfyui is a nightmare and and beautifull dream at the same time... every ting is open-endend and complex... but highly adaptable and solid at the same time... once you got a workflow, you keep it...  Be sure to get the comfyui manager as quickly as your first step. The community is making all new nodes... and this manager will help you install those and als models from a repository... So once you are over this first hurdle it will grow on you! 

Thanks for making a cool upscaler!  I teach ai-art to designers and photographers.. they all are very pleased with it... 


[Upscale workflow.json](https://github.com/upscayl/upscayl/files/13842148/Upscale.workflow.json)

**Reply by GRwald** [reply]
oh yes there are opensource developed sjizzle to make web apps on it so you can run it online like you are planning to do with Upscayle! wuick search on reddit and you will find it!

**Reply by furdiburd** [reply]
The ultrasharp model does something like that but not sure if that is what you want

**Comment by Woisek** [comment]
Regarding 'add detail', I want to share my experience: I mostly generate in 724 x 1024 when using SDXL and then upscale 2 times 2x to get 4k images.
Now what I have experienced, is, that with upscayl, the first 2x upscale often does a great job. But upscaling the upscaled image again 2x gives a horrible result, no matter what upscaler is used.
And I also noticed that upscayl has massive problems upscaling color gradients. This is probably due to denoise, but setting denoise too low, doesn't add sufficient more details to the image, so ... some balance has to be found or something that evens this out.

Other than that, it could be a serious competitor against Gigapixel AI.

**Reply by aaronliu0130** [reply]
You shouldn't be upscayling it 2 times by 2x. The native output of most models is 4x, and upscayl shrinks that down to your preferred output resolution in post-processing.
Also, don't give us the credit for the image quality 😅, we're merely a GUI for REAL-ESRGAN.

**Reply by Woisek** [reply]
> You shouldn't be upscayling it 2 times by 2x. The native output of most models is 4x, and upscayl shrinks that down to your preferred output resolution in post-processing.

Yepp, I know that, but first, I have only 8GB of VRAM, so I have to tread lightly, and second, when doing this with a smaller image size, that my GraCa can handle, a 4x upscale was almost never a good upscale. That's why I took the 2 times 2x upscale with incredible results even up to 8k in ComfyUI.
I think the real drawback is that we can't use prompts like we are used to in A1111/SD:Next and ComfyUI when doing upscales.

> Also, don't give us the credit for the image quality 😅, we're merely a GUI for REAL-ESRGAN.

Sure, but how would we use REAL-ESRGAN easily if we wouldn't have this wonderful GUI? 🙂 Thanks for this. 👍

**Reply by aaronliu0130** [reply]
Well, be warned, selecting 2x takes more processing power (and probably VRAM) than just selecting 4x as it's an extra step. Like I said, you'd be better off just Upscayling 2x. I also have only 8G of VRAM, it works nicely 😀

**Reply by Woisek** [reply]
Hmmm ... right now I can't get behind why this needs more power and is an extra step, but more important is, that I found out, that an upscale straight to 4x delivers more than often a very ugly result. That's another reason why I do 2 times 2x upscale. 
But I'm open to suggestions on how to resolve this.

**Reply by GRwald** [reply]
Yes! On the other hand with an upscale of 2x with diffusion, add detail lora, conrolnet tiling and esrgan2x as above... and then pull it through Upscayl with ultrasharp4x I managed to put a painterly line drawing with flowers and reed to get over 10.000pixels! the jpg was 140mb at the end... very pleased with the end result at 100% zoom even

**Comment by timohenz** [comment]
Seconded this. Looking at the r/stablediffusion subreddit seems like people are reaching the next level in upsacling. I have tried but am not smart enough to get one of these tutorials with tiles to run in auto1111.

**Reply by aaronliu0130** [reply]
(Unfortunately?) We don't use Stable Diffusion.

**Reply by NayamAmarshe** [reply]
Yeah, Krea is not an upscaler. It's a detail-painter. What Upscayl is doing is actual upscaling, for free and for all :)

**Comment by PierrunoYT** [comment]
It would be cool there is a mod for a image enhance like krea or magnific
